vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go

```diff
@@ -55,18 +55,25 @@ func GetContainersResources(pod *core.Pod, podRecommendation vpa_types.Recommend
 			klog.V(2).Infof("no matching recommendation found for container %s", container.Name)
 			continue
 		}
+
+		containerPolicy := vpa_api_util.GetContainerResourcePolicy(container.Name, vpaResourcePolicy)
```

Comment:
How about adding a helper function GetContainerLimitPolicy? It could look for a policy and return default one in case none is found.
Then the if in line 67 will be simplified to if limitPolicy == "Auto"